iroh-net/src/endpoint.rs

```diff
@@ -535,17 +535,40 @@ impl Endpoint {
     ///
     /// # Errors
     ///
-    /// Will return an error if we attempt to add our own [`PublicKey`] to the node map.
+    /// Will return an error if we attempt to add our own [`PublicKey`] to the node map or if the
+    /// direct addresses are a subset of ours.
     pub fn add_node_addr(&self, node_addr: NodeAddr) -> Result<()> {
+        self.add_node_addr_inner(node_addr, magicsock::Source::App)
+    }
+
+    /// Informs this [`Endpoint`] about addresses of the iroh-net node, noting the source.
+    ///
+    /// This updates the local state for the remote node.  If the provided [`NodeAddr`] contains a
+    /// [`RelayUrl`] this will be used as the new relay server for this node.  If it contains any
+    /// new IP endpoints they will also be stored and tried when next connecting to this node. The
+    /// source is used for logging exclusively and will not be stored.
+    ///
+    /// # Errors
+    ///
+    /// Will return an error if we attempt to add our own [`PublicKey`] to the node map or if the
+    /// direct addresses are a subset of ours.
```

Comment:
When i first read this i assumed it would error as soon as one of the added addresses is also one of ours.  But the implementation silently removes those instead.  I know the wording here is technically correct, but maybe there's a different way of wording this.

```
or if the direct addresses contains only addresses known to belong to
this [`Endpoint`] already.
```
maybe.

Maybe the whole pruning should be described a bit as well, e.g.:
```
It is common that nodes might have the same direct addresses which are
only valid in their local network, while the nodes are not in the same
network.  Because of this direct addresses known to belonging to this
node will be silently removed from the addresses added.
```

nit: I don't think we should use a style like "ours" in the docs, though I often find myself writing like that as well.

Reply:
done in c3c95c1d

I'll avoid the ours in the future. In the meantime this is simply following the style of docs around the place